profiles/common/modules/features/nexteuropa_dgt_connector/tmgmt_dgt_connector/tmgmt_dgt_connector_cart/includes/forms/tmgmt_dgt_connector_form_cart_bundle_items.inc

```diff
@@ -0,0 +1,292 @@
+
+  // Refreshing the form.
+  if (empty($form_state['redirect'])) {
+    $form_state['rebuild'] = TRUE;
+  }
+
+  // Prevent running reload function if we are out of the ajax context.
+  if (isset($form_state['ajax']) && $form_state['ajax']) {
+    ctools_ajax_command_reload();
+  }
+}
+
+/**
+ * Updates the CartBundle entity properties.
+ *
+ * @param array $form_state
+ *   The form state array.
+ */
+function _tmgmt_dgt_connector_cart_update_cart_bundle(array &$form_state) {
+  // Loading the CartBundle entity.
+  $cbid = $form_state['build_info']['args']['cbid'];
+  $cart_bundle = entity_load_single('cart_bundle', $cbid);
+
+  // Updating the change date of a CartBundle entity.
+  $cart_bundle->save();
+
+  // Discard the CartBundle entity if needed and close the popup.
+  if (_tmgmt_dgt_connector_cart_discard_cart_bundle($form_state, $cart_bundle)) {
+    $form_state['redirect'] = TMGMT_DGT_CONNECTOR_CART_VIEW_PATH;
+    $form_state['ajax_commands'][] = ctools_modal_command_dismiss();
+    $form_state['ajax_commands'][] = ctools_ajax_command_redirect(TMGMT_DGT_CONNECTOR_CART_VIEW_PATH);
+  }
+}
+
+/**
+ * Updates CartItem entities properties.
+ *
+ * @param array $form_state
+ *   The form state array.
+ */
+function _tmgmt_dgt_connector_cart_update_cart_items(array &$form_state) {
+  // Getting the CartItems IDs.
+  $ids = array_keys($form_state['values']['cart_items_fields']);
+  $fields = $form_state['values']['cart_items_fields'];
+  $cart_items = entity_load('cart_item', $ids);
+
+  // Iterating through the CartItem entities and setting their properties.
+  foreach ($cart_items as $id => $cart_item) {
+    foreach ($fields[$id] as $field => $value) {
+      $cart_item->{$field} = check_plain($value);
+    }
+    $cart_item->save();
+  }
+}
+
+/**
+ * Discards selected CartItems and a CartBundle if needed.
+ *
+ * @param array $form_state
+ *   The form state array.
+ * @param \Drupal\tmgmt_dgt_connector_cart\Entity\CartBundle $cart_bundle
+ *   The CartBundle entity.
+ *
+ * @return bool
+ *   Returns TRUE if the CartBundle entity was discarded else FALSE.
+ */
+function _tmgmt_dgt_connector_cart_discard_cart_bundle(array &$form_state, CartBundle $cart_bundle) {
+  if ($form_state['values']['op'] == TMGMT_DGT_CONNECTOR_CART_SUBMIT_DISCARD) {
+    $selected_items = _tmgmt_dgt_connector_cart_get_selected_cart_items($form_state);
+    if ($selected_items == $form_state['values']['cart_items_selected']) {
+      $cart_bundle->status = CartBundle::STATUS_DISCARDED;
+      $cart_bundle->save();
+
+      return TRUE;
+    }
+  }
+
+  return FALSE;
+}
+
+/**
+ * Returns selected CartItem IDs based on the form state.
+ *
+ * @param array $form_state
+ *   The form state array.
+ *
+ * @return array
+ *   An array with IDs of selected CartItems entities.
+ */
+function _tmgmt_dgt_connector_cart_get_selected_cart_items(array &$form_state) {
+  return array_filter($form_state['values']['cart_items_selected'], function ($value) {
```

Comment:
There is no need to create this anonymous function here.

See: https://3v4l.org/Lpm04